cmd/disk-cache-backend.go

```diff
@@ -402,14 +468,29 @@ func (c *diskCache) Put(ctx context.Context, bucket, object string, data io.Read
 		bufSize = size
 	}
 
-	n, err := c.bitrotWriteToCache(ctx, cachePath, data, size)
+	var metadata = make(map[string]string)
+	for k, v := range opts.UserDefined {
+		metadata[k] = v
+	}
+	var reader = data
+	var actualSize = uint64(size)
+	var err error
+	if globalCacheKMS != nil {
```

Comment:
the cache encryption reader needs to be wrapped around the reader only if the cache encryption master key is set, which is why this check is in place. Otherwise the reader is passed down for bitrot write.